src/vm/wren_compiler.c

```diff
@@ -586,6 +586,21 @@ static void readNumber(Parser* parser)
     nextChar(parser);
     while (isDigit(peekChar(parser))) nextChar(parser);
   }
+  
+  // See if the number is in scientific notation
+  if (peekChar(parser) == 'e' || peekChar(parser) == 'E')
+  {
+    nextChar(parser);
+    
+    // if the exponant is negative
+    if(peekChar(parser) == '-')
+      nextChar(parser);
+    
+    if(!isDigit(peekChar(parser)))
+      lexError(parser, "Unterminated scientific notation.");
```

Comment:
And this to:

```c
if (!isDigit(peekChar(parser)))
{
  lexError(parser, "Unterminated scientific notation.");
}
```

My style rules here are:

1. Always have a space after `if` (and other control flow keywords).
2. Only omit the `{}` on an `if` if the entire thing fits on one line.

Thanks!